[GeoMechanicsApplication] Fixed occasional crash when constructing `LineInterfaceGeometry` instances

avdg81 opened the pull request:
**📝 Description**

When a `LineInterfaceGeometry` instance was constructed with a list of node pointers where at least one of them is null, the application would crash. The underlying reason was that calculating the points of the mid-line geometry failed, due to dereferencing the null pointer. This fix checks whether there are any null pointers in the provided vector of node pointers, and if there are it will return a vector of null pointers for the mid-line points.

Note that being able to deal with such input is required, since at element registration time we always provide the "blueprint" element with a list of null pointers for the element's nodes.

avdg81 committed:
The constructor of `LineInterfaceGeometry` now supports lists of node pointers that contain at least one null pointer rather than crashing

rfaasse commented on applications/GeoMechanicsApplication/custom_geometries/line_interface_geometry.h:
```diff
@@ -240,6 +242,13 @@ class LineInterfaceGeometry : public Geometry<Node>
+            // At least one point is not defined, so the points of the mid-line can't be computed.
+            // As a result, all the mid-line points will be undefined.
+            return PointerVector<Node>{number_of_midline_nodes};
+        }
```
Ah, we missed the .get(), nice way of checking!

avdg81 replied:
In fact, we don't need the call to `get()` to check whether or not it is a null pointer, since we can use `operator==` directly. I've removed the redundant member function call.

FYI: What we had done wrong previously, was to use `begin()` and `end()`, rather than `ptr_begin()` and `ptr_end()`.

rfaasse commented on applications/GeoMechanicsApplication/tests/cpp_tests/custom_geometries/test_interface_geometry.cpp:
```diff
@@ -61,6 +61,18 @@ KRATOS_TEST_CASE_IN_SUITE(InterfaceGeometryIsAGeometry, KratosGeoMechanicsFastSu
+    const auto geometry = LineInterfaceGeometry<Line2D3<Node>>{six_null_pointers_to_nodes};
+
+    KRATOS_EXPECT_EQ(geometry.PointsNumber(), 6);
+    KRATOS_EXPECT_EQ(geometry.WorkingSpaceDimension(), 2);
```
We could add LocalSpaceDimension here, but I don't think it matters too much

avdg81 replied:
What a nice suggestion. I've added it here and also to two more test cases.

rfaasse reviewed (dismissed):
Looks good, nicely scoped and well tested little PR. Nothing blocking to add, just a minor suggestion

avdg81 committed:
Processed Richard's review feedback

- Removed an unnecessary call to member function `get()`.
- Added a few assertions that check whether the local space dimension equals one.

rfaasse reviewed (approved):
Good to go! Thanks for incorporating the suggestions!
